v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SpannerTransactionWriterDoFn.java

```diff
@@ -196,11 +176,16 @@ public void processElement(ProcessContext c) {
       if (retryCount != null) {
         isRetryRecord = true;
       }
+
+      ChangeEventSessionConvertor csc =
+          new ChangeEventSessionConvertor(
+              schema, transformationContext, sourceType, roundJsonDecimals);
```

Comment:
Does this need to created per element in `@ProcessElement` or can this be done in the Setup phase?
Creating a new object per event in `@ProcessElement` looks to be very expensive?

Reply:
Agreed, thats a good idea.